koa-crud/src/data-access/mongoose/promo-enrollment-requests/actions.ts

```diff
@@ -1,35 +1,59 @@
-import { PromoEnrollmentRequest as PromoEnrollmentRequestModel } from '../../../lib/mongoose/models/promo-enrollment-request';
-import { PromoEnrollmentRequestsStore } from '../../../types/index';
-
-const actions = ({
+import PromoEnrollmentRequestModelType, {
   PromoEnrollmentRequest,
+  PromoEnrollmentRequestDocument,
+} from '../../../lib/mongoose/models/promo-enrollment-request';
+
+type PromoEnrollmentRequestFilters = {
+  _id?: PromoEnrollmentRequest['_id'] | Record<string, any>;
+  promo?: PromoEnrollmentRequest['promo'];
+  member?: PromoEnrollmentRequest['member'];
+};
```

Comment:
You can use Partial and Pick combo here.

Reply:
noted on this sir